packages/server/src/controllers/post/filterPost.ts

```diff
@@ -21,93 +23,155 @@ import logger from "../../utils/logger";
 import error from "../../errorResponse.json";
 import { GET_POSTS_FILTER_COUNT } from "../../constants";
 
+const querySchema = z.object({
+  first: z.coerce
+    .string()
+    .transform((value) => parseAndValidateLimit(value, GET_POSTS_FILTER_COUNT)),
+  page: z.coerce
+    .string()
+    .optional()
+    .transform((value) => (value ? parseAndValidatePage(value) : undefined)),
+  limit: z.coerce
+    .string()
+    .optional()
+    .transform((value) => parseAndValidateLimit(value, GET_POSTS_FILTER_COUNT)),
+  after: z.uuid().optional(),
+  created: z.enum(["ASC", "DESC"]).default("DESC"),
+});
+
 type ResponseBody = IFilterPostResponseBody | IApiErrorResponse;
 
 export async function filterPost(
   req: Request<unknown, unknown, IFilterPostRequestBody>,
   res: Response<ResponseBody>,
 ) {
+  if (req.query?.page || req.query?.limit) {
+    logger.warn(
+      "Offset-based pagination is deprecated and will be removed in next major release. Please migrate to cursor pagination instead.",
+    );
+  }
+
+  const query = querySchema.safeParse(req.query);
+  if (!query.success) {
+    return res.status(400).json({
+      code: "VALIDATION_ERROR",
+      message: "Invalid query parameters",
+      errors: query.error.issues,
+    });
+  }
+
+  const { first: _first, page, after, created, limit } = query.data;
+  const first = req.query?.limit ? limit : _first;
```

Comment:
@Vedant005 There is a breaking change over here.

Previously, we were accessing the `page` and `limit` from `req.body`.